Documents/Command/AWS-RunAnsible/AWS-RunAnsible.yaml

```diff
@@ -0,0 +1,189 @@
+      >&2'
+    - 'if cat /etc/issue|grep -i Amazon ; then '
+    - '   sudo pip install ansible --upgrade'
+    - '   sudo pip install awscli --upgrade'
+    - 'elif cat /etc/issue|grep -i Ubuntu ; then '
+    - '   sudo apt-get update'
+    - '   sudo DEBIAN_FRONTEND=noninteractive apt-get install python-pip -y'
+    - '   sudo pip install ansible --upgrade'
+    - '   sudo DEBIAN_FRONTEND=noninteractive apt-get install unzip -y'
+    - '   sudo DEBIAN_FRONTEND=noninteractive apt-get install awscli -y'
+    - 'elif cat /etc/issue|grep -i ''\\S'' ; then '
+    - '   sudo yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm'
+    - '   sudo yum -y update'
+    - '   sudo yum -y install python-pip'
+    - '   sudo pip install ansible --upgrade'
+    - '   sudo yum -y install wget'
+    - '   sudo yum -y install unzip'
+    - '   sudo pip install awscli --upgrade'
+    - else
+    - '   echo "There was a problem installin or updating the required tools for the
+      document. Plesae review the logs and correct the problem" >&2'
+    - '   exit 1'
+    - fi
+    - 'if [ -z "{{WorkingDirectory}}" ] ; then '
+    - '  cd `mktemp -d`'
+    - else
+    - '  mkdir -p {{WorkingDirectory}}; cd {{WorkingDirectory}}'
+    - fi
+    - echo "Running Ansible in `pwd`"
+    - '#if [ -z "${PlaybookFile}" ]'
+    - '#then'
+    - '#   PlaybookFile="ssm-playbook.yml"'
+    - '#else'
+    - PlaybookFile="{{PlaybookFile}}"
+    - '#fi'
+    - if [ -z "{{playbook}}" ] ; then
+    - '   PlaybookURL="{{PlaybookURL}}"'
+    - '   filename=${PlaybookURL##*/}'
+    - '   extension=${filename##*.}'
+    - '   if [[ "${PlaybookURL}" == http* ]]; then'
+    - '   if [[ "{{PlaybookURL}}" == */ ]]; then'
+    - '     echo "http URL is only supportted for compressed files." >&2'
+    - '     exit 1'
+    - '   fi'
+    - '#-------------- added for zip support---------------'
+    - '# If we have an extensionension, assume we have a single file and copy it;
+      otherwise, sync'
+    - '   wget ${PlaybookURL}'
+    - '   if [ -n "$extension" ] ; then'
+    - '     case "$extension" in'
+    - '      zip) unzip ${filename}'
+    - '        ;;'
+    - '      tgz | gz | bz2 | tbz2 | tar) tar xf ${filename}'
+    - '        ;;'
+    - '      yml | yaml) if ! cmp --silent ${filename} ${PlaybookFile};then'
+    - '           cp ${filename} ${PlaybookFile}'
+    - '        fi'
+    - '        ;;'
+    - '      *) echo "Unsupported compression format"'
+    - '        exit 1'
+    - '        ;;'
+    - '     esac'
+    - '  fi'
+    - '#---------------added for zip support --------------'
+    - '  if [ $? -ne 0 ]; then'
+    - '       echo "There was a problem downloading the playbook. Make sure the URL
+      is correct and that the playbook exists." >&2'
+    - '       exit 1'
+    - '   fi'
+    - '  elif [[ "${PlaybookURL}" == s3* ]] ; then'
+    - '# If we have an extensionension, assume we have a single file and copy it;
```

Comment:
The original code runs an "aws --version" and checks the $? value to ensure the AWS CLI is installed.